Sources/SwiftDocC/Infrastructure/Input Discovery/DocumentationInputsProvider.swift

```diff
@@ -183,32 +183,36 @@ extension DocumentationContext.InputsProvider {
         let derivedDisplayName = moduleNames.count == 1 ? moduleNames.first : nil
 
         let info = try DocumentationContext.Inputs.Info(bundleDiscoveryOptions: options, derivedDisplayName: derivedDisplayName)
-
-        var topLevelPages: [URL] = []
-        if moduleNames.count == 1, let moduleName = moduleNames.first, moduleName != info.displayName {
-            let tempURL = fileManager.uniqueTemporaryDirectory()
-            try? fileManager.createDirectory(at: tempURL, withIntermediateDirectories: true, attributes: nil)
-
-            let url = tempURL.appendingPathComponent("\(moduleName).md")
-            topLevelPages.append(url)
-            try fileManager.createFile(
-                at: url,
-                contents: Data("""
+        
+        let topLevelPages: [URL]
+        let provider: DocumentationBundleDataProvider
+        if moduleNames.count == 1, let moduleName = moduleNames.first, moduleName != info.displayName, let url = URL(string: "in-memory-data://\(moduleName).md") {
```

Comment:
The "in-memory-data" protocol seems like a very different implementation than what we had before. What's the overall idea here?